client/src/common/components/Members/components/MemberDetails.jsx

```diff
@@ -174,8 +178,9 @@ class MemberDetails extends PureComponent {
           </button>
           <SwitchButton
             checked={checked}
+            disabled={disabled}
             label={label}
-            onChange={this.handleChangingRoles}
+            onChange={disabled ? null : this.handleChangingRoles}
```

Comment:
try `!disabled && this.handleChangingRoles`

Reply:
I really tried, I started from your solution, there is a react error that onChange should receive function or undefined/null if this adding of handler depends on other props...